### ISSUE TITLE
Feature Request: Supports image server URL which return multipart (stream) contents (not raw blob data)

### ISSUE THREAD

**Issue by dreampiggy** [SDWebImage/SDWebImage#3033]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 12.0 / 10.14.0 / 12.0 / 5.0
 SDWebImage Version      | e.g. 5.0.0 / 4.4.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 11 / Xcode 10
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Current SDWebImage only assume the URLSession download task returns the full Raw Data. Which the content type is mostly `image/jpeg`, `image/png`, etc.

However, some modern Webside may use multipart like `multipart/mixed` to return the data. This on URLSession, will become `NSURLSessionStreamTask`.

For example, this URL: `https://dummyimage.com/600x400/000/fff`, can loaded on Chrome, on React-Native, but not on current SDWebImage.

We can reference the implementation from React-Native here:

https://github.com/facebook/react-native/blob/0.63-stable/React/Base/RCTMultipartDataTask.m#L101-L128

**Comment by dreampiggy** [comment]
I have a idea that using the NSURLSessionStreamTask can solve the problem in #2353

We use the Stream Task, and [mmap](https://man7.org/linux/man-pages/man2/mmap.2.html), to create a memory-mapped-file on the disk path.

The once the stream task new data available, we append that buffer to the mmaped memory buffer.

Even if the RAM usage is high, the system will try to purge down and page out the extra memory for using. Because the mmaped memory is clean memory, not dirty memory like normal `malloc`

Have a try to works on this @Insofan
